File: solutions/go/difference-of-squares/1/difference_of_squares.go
package diffsquares

// SquareOfSum function returns the sum of _n_ first integers squared
func SquareOfSum(n int) int {
	sum := n * (n + 1) / 2
	return sum * sum
}

// SumOfSquares function returns the sum of squares of _n_ first integers
func SumOfSquares(n int) int {
	return (n * (n + 1) * (2*n + 1)) / 6
}

// Difference function returns the difference between the square of sum and the sum of squares of the _n_ first integers
func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}

File: solutions/clojure/difference-of-squares/2/src/difference_of_squares.clj
(ns difference-of-squares)

(defn square-of-sum [N] (let [s (* N (inc N) 1/2)] (* s s)))
(defn sum-of-squares [N] (* N (inc N) (+ 1 N N) 1/6))
(defn difference [N] (- (square-of-sum N) (sum-of-squares N)))
